internal/sync/controller.go

```diff
@@ -78,12 +78,14 @@ func (p Controller) SynchroniseTimeframe(ctx context.Context, start time.Time, e
 		return err
 	}
 
-	// remove declined events
-	if !p.SyncDeclinedEvents {
-		log.Debug("We're not syncing declined events, to enable set sync.sync_declined_events to true in your config.yaml")
-		eventsInSource = removeDeclinedEvents(eventsInSource)
+	filteredEventsInSource := []models.Event{}
+
+	for _, filter := range p.filters {
+		p.logger.Info("loaded filter", "name", filter.Name())
 	}
 
+	filteredEventsInSource = FilterEvents(eventsInSource, p.filters...)
```

Comment:
I wonder whether the filters should use the same loop structure as for transforming events. The outer loop iterates over each event and the inner through each Transformer/Filter.  Either variant works, but passen each event to each filter individually, would reduce the code duplication.

Reply:
I'm not sure about what you mean. Do you propose to handle the filter and transformations in the same “big” for loop which first handles filtering and then (optionally) handles the transformation? 

>  but passen each event to each filter individually, would reduce the code duplication.

i thought about this as well, there's two things which made me filter all the events at once, not one by one:

- First, if we filter a single event and want to hand it over to the transformer in the next step we need to rethink the Filter interface. Maybe the `Filter` func could return something like a bool (true if filtered and false if still in)
  - we then could filter the event, based on the returned bool we can then transform and add the event to the list of to be synced events, okay cool! 
  - but this would add some more nesting (at least what i'm having in my head) in the SynchroniseTimeframe function, which i'm not a fan of.
- In my approach though, we create a new list with only those events which are relevant and transform those. the steps are clearly separated. 


Let us have a vCoffee next week, if you're free and we can go over the different approaches :)